calcite actionbar implementation

mrwillyees opened the pull request:
implementation of the calcite actionbar using toolbar, toolbutton and toolseparator.
atm using the toolseparator as divisor between groups of actions as it can be seen here:
![image](https://user-images.githubusercontent.com/91129623/151554050-fb00d97a-c19c-40fa-8c4b-30dbee4fef4d.png)
Quick demo showing difference between material implementation and calcite. to be noted is the not automatic fillwidth of a button to the parent (or parent layout), they need to be set manually for every button with `Layout.fillWidth : true`. Also, removed the backgorund `Rectangle` from the `ToolBar`
https://user-images.githubusercontent.com/91129623/151554530-8f9a1273-1f96-4a71-9fb3-8ab48173f04a.mp4

anmacdonald commented on calcite-qml/Calcite/ToolBar.qml:
```diff
@@ -0,0 +1,34 @@
+    implicitHeight: Math.max(implicitBackgroundHeight + topInset + bottomInset,
+                             contentHeight + topPadding + bottomPadding)
+    spacing: 16
+    background: Rectangle {
+        implicitHeight: 48
+        color: "transparent"
+    }
+}
```
🔧 One thing to note is that in your current ToolBar implementation, there may be visible gaps between buttons unless they are squished tightly together explicitly with `spacing: 0`:

<img width="227" alt="image" src="https://user-images.githubusercontent.com/42203545/152179249-65a56cae-570a-4b71-93cd-c23a7a58a546.png">

Based on JS FloorFilter / Figma, / Material Style, I  think you should have a filled background, as opposed to being transparent.

mrwillyees replied:
Fixing this by using the same color used in the `ToolBar` background.
An additional change could be to use by default `spacing : 0` since I cannot see any spacing in the specifications.

anmacdonald commented on calcite-qml/Calcite/Calcite.qml:
```diff
@@ -43,18 +43,28 @@ QtObject {
 
     readonly property color background: theme === Calcite.Theme.Light ? offWhite : "#353535"
 
+    readonly property color background2: theme === Calcite.Theme.Light ? "#C7EAff" : "#151515"
+
+    readonly property color background3: theme === Calcite.Theme.Light ? "FFFFFF" : "#404040"
+
     readonly property color foreground1: theme === Calcite.Theme.Light ? "#FFFFFF" : "#2B2B2B"
 
     readonly property color foreground2: theme === Calcite.Theme.Light ? "#F3F3F3" : "#202020"
 
     readonly property color foreground3: theme === Calcite.Theme.Light ? "#EAEAEA" : "#151515"
 
+    readonly property color foreground4: theme === Calcite.Theme.Light ? "#004874" : "#59d6ff"
+
+    readonly property color foreground5: theme === Calcite.Theme.Light ? "#EAEAEA" : "#2B2B2B"
+
     readonly property color text1: theme === Calcite.Theme.Light ? "#151515" : "#FFFFFF"
 
     readonly property color text2: theme === Calcite.Theme.Light ? "#4A4A4A" : "#BFBFBF"
 
     readonly property color text3: theme === Calcite.Theme.Light ? "#6A6A6A" : "#9F9F9F"
 
+    readonly property color text4: theme === Calcite.Theme.Light ? "#2B2B2B" : "#FFFFFF"
+
```
🔧  We had a chat on restricting this list of var to: `https://github.com/Esri/calcite-colors/blob/master/src/colors.ts#L832`.

I am not against updating the existing color profiles to be conformant with figma/react.  But it is worth noting that as it currently stands,  that the ToolButtons look different in terms of highlighting behavior and text color to the rest of the available controls.

<img width="495" alt="image" src="https://user-images.githubusercontent.com/42203545/152181151-56448380-8c28-4766-bbac-61a85d3510aa.png">

Looping in @nCastle1 as he might have an idea of what the most "correct" color scheme is.

anmacdonald commented on calcite-qml/Calcite/ToolSeparator.qml:
```diff
@@ -0,0 +1,36 @@
+T.ToolSeparator {
+    id: control
+
+    padding: 0
```
❓  Is this needed?

mrwillyees replied:
not needed, removed

anmacdonald reviewed (changes requested):
👍 ToolButtons are looking great. There's a general question on conformity to the rest of the app that we'll need to think about.

Please remember to update **CHANGELOG.md` as part of this commit, and the Calcite README.

mrwillyees commented:
reminder status: to be decided the style to use on the `ToolButton`, either the blue background used in the JS demos or the basic calcite (figma) `Action` component.

mrwillyees commented:
Implemented the blue background to be shown when `ToolBar.flat === true`, removed the new colors not Calcite standard and storing them into the tool directly

anmacdonald commented on calcite-qml/Calcite/ToolButton.qml:
```diff
@@ -0,0 +1,66 @@
+        font: control.font
+
+        color: control.icon.color
+        alignment: control.display === AbstractButton.TextBesideIcon ? Qt.AlignLeft : Qt.AlignCenter
```
❌  Looking at this in retrospect - the text no longer aligns properly.

<img width="299" alt="image" src="https://user-images.githubusercontent.com/42203545/154306450-2eda4931-1d0e-488a-964c-9ecb733fdbd0.png">

Left alignment is possible the wrong way to go about this. Either we keep everything centered or we try something like:

```qml
alignment: control.display === AbstractButton.TextBesideIcon && icon.source.toString() !== "" && text != "" ? Qt.AlignLeft : Qt.AlignCenter
```

Thoughts?

anmacdonald reviewed (changes requested):
👍 Getting there. More to say on color scheme.

mrwillyees commented on calcite-qml/Calcite/ToolBar.qml:
```diff
@@ -0,0 +1,36 @@
+
+    background: Rectangle {
+        implicitHeight: 48
+        color: Calcite.foreground1
```
The `ToolBar` is the container of the buttons. The Calcite respective should be the `Action Bar`. From the Figma, I can see that it follows the idle colors of the buttons. From the screenshot, I believe that long spacing is not a button, but the below `ActionBar`, so I m incline to follow idle colors used for calcite (not flat). Does it answer the query?
![image](https://user-images.githubusercontent.com/91129623/154475511-1760b4eb-f927-406a-8a40-542a25e13ecf.png)
![image](https://user-images.githubusercontent.com/91129623/154475931-6414ee72-87a0-460d-b986-0c906ff5bc91.png)

anmacdonald replied:
I'm afraid I'm not clear on this one. Are you advocating for using a foreground color as a background color?

anmacdonald reviewed (approved):
👍 Thanks for the changes and talking it through with me.
